### PR TITLE
[Feature]Add Vit

### PR DESCRIPTION
**Pull request by HIT-cwh**
Hi,

This pr currently provide the following features:

1. Train our own ViT on ImageNet 1K w/o AutoAugment.
2. The model of ViT, including the ViT backbone, ViT classification head, etc.
3. Some configs for ViT training, finetune, and inference.
4. Pretrained weights from pytorch-image-models (after converting the keys to fit our own model) can be used for inference on ImageNet and get the same accuracy reported in pytorch-image-models.

Some works need to do:

1. Train our own ViT on ImageNet 1K w/ AutoAugment.
2. Finetune our own Vit on ImageNet 1K.

Here’re some references:

1. Basic architecture: Please refer to pytorch-image-models.
2. Initializations for each layers: Please refer to ClassyVision.
3. Training configs: Please refer to ClassyVision.
4. Names and structures of modules: Please refer to MMDet.

Hope this can help you. If there’s any question, please let me know.

Best regards
whcao

### CODE REVIEW

File: configs/_base_/schedules/imagenet_bs4096.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# optimizer
+optimizer = dict(
+    type='AdamW', lr=0.0015,
+    weight_decay=0.3)  # 0.003 / 4096 * (32bs * 64gpus) = 0.0015
```
Comment (ycxioooong) [inline configs/_base_/schedules/imagenet_bs4096.py:4]: The lr configuration is for batch_size = 2048. So (i) we'd better rename the filename as imagenet_bs2048.py; (ii) it's better to explain more about the equation 0.003 / 4096 * (32bs * 64gpus) = 0.0015; maybe add some text to explain this.

Comment (ycxioooong) [inline configs/_base_/schedules/imagenet_bs4096.py:4]: The lr configuration is for batch_size = 2048. So (i) we'd better rename the filename as imagenet_bs2048.py; (ii) it's better to explain more about the equation 0.003 / 4096 * (32bs * 64gpus) = 0.0015; maybe add some text to explain this.

File: tests/test_backbones/test_vision_transformer.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import pytest
+import torch
+from torch.nn.modules import GroupNorm
+from torch.nn.modules.batchnorm import _BatchNorm
+
+from mmcls.models.backbones import VisionTransformer
+
+# from mmcls.models.backbones.vision_transformer import
+# FFN, MultiheadAttention, TransformerEncoderLayer
```
Comment (ycxioooong) [inline tests/test_backbones/test_vision_transformer.py:9]: these comments could be deleted

**Review by ycxioooong — CHANGES_REQUESTED** [review]

**Review by ycxioooong — APPROVED** [review]
LGTM
